packages/list-store/src/use-searchable-facet.tsx

```diff
@@ -0,0 +1,227 @@
+  if (!storeRef.current) {
+    storeRef.current = createFacetStore(props);
+  }
+  return (
+    <FacetContext.Provider value={storeRef.current}>
+      {children}
+    </FacetContext.Provider>
+  );
+}
+
+interface GetLetterCategoriesProps {
+  filters: SearchableFilter[];
+  letterCategory?: string;
+  searchValue?: string;
+}
+
+export function getLetterCategories({
+  filters,
+  letterCategory,
+  searchValue,
+}: GetLetterCategoriesProps) {
+  const filtersFilteredBySearchValue = filterBySearchValue(
+    filters,
+    searchValue
+  );
+
+  return [
+    ...new Set([
+      // Always include the letter category of the current filter
+      ...(letterCategory ? [letterCategory] : []),
+      ...filtersFilteredBySearchValue
+        .map(filter => filter.letterCategory)
+        .filter(category => !!category),
+    ]),
+  ].sort();
+}
+
+export function extendFiltersWithLetterCategory(filters?: Filter[]) {
+  if (!filters?.length) {
+    return [];
+  }
+  return filters.map(filter => {
+    let letterCategory = '';
+    const name = filter.name?.toString();
+    if (name) {
+      const firstLetter = name.toUpperCase().replace(/[^0-9A-Z]/g, '')[0];
+      if (/^\d+$/.test(firstLetter)) {
+        letterCategory = '[0-9]';
+      } else {
+        letterCategory = firstLetter;
+      }
+    }
+    return {
+      ...filter,
+      name: name || '',
+      letterCategory,
+    };
+  });
+}
+
+interface GetFilteredFiltersProps {
+  filtersWithLetterCategory: SearchableFilter[];
+  searchValue: string;
+  letterCategory: string;
+  sortBy: FacetSortBy;
+}
+
+function filterBySearchValue(
+  filters: SearchableFilter[],
+  searchValue?: string
+) {
+  if (searchValue) {
+    return filters.filter(
+      filter => filter.name?.toLowerCase().includes(searchValue.toLowerCase())
+    );
+  } else return filters;
+}
+
+function filterByLetterCategory(
+  filters: SearchableFilter[],
+  letterCategory: string
+) {
+  if (letterCategory) {
+    return filters.filter(filter => filter.letterCategory === letterCategory);
+  } else return filters;
+}
+
+export function getFilteredFilters({
+  filtersWithLetterCategory,
+  searchValue,
+  letterCategory,
+  sortBy,
+}: GetFilteredFiltersProps) {
+  const filtersFilteredBySearchValue = filterBySearchValue(
+    filtersWithLetterCategory,
+    searchValue
+  );
+
+  const filtersFilteredByLetterCategory = filterByLetterCategory(
+    filtersFilteredBySearchValue,
+    letterCategory
+  );
+
+  if (sortBy === 'alphabetical') {
+    return filtersFilteredByLetterCategory.sort((a, b) => {
+      if (a.name && b.name) {
+        return a.name.localeCompare(b.name);
+      }
+      return 0;
+    });
+  } else {
+    return filtersFilteredByLetterCategory.sort((a, b) => {
+      return b.totalCount - a.totalCount;
+    });
+  }
+}
+
+export function useSearchableMultiSelectFacet() {
+  const storeContext = useContext(FacetContext);
+  if (!storeContext) throw new Error('Missing FacetContext in the tree');
```

Comment:
This is perfectly valid code, but for consistency with other parts of the codebase I prefer the use of curly brackets:

```
if (!storeContext) {
   // ...
}
```